Repair not accessible links (4xx status)

arbox opened the pull request:
See #17 for details.

arbox committed:
Repair not accessible links (4xx status)

chrisseaton commented on index.md:
```diff
@@ -181,7 +181,7 @@ In the IPSJ Journal of Programming (PRO), 2005. *In Japanese*.
 <span class="badge badge-warning">TruffleRuby</span>
 
-<a class="paper" name="seaton2014" href="#seaton2014">#</a> C. Seaton, M. L. Van De Vanter, M. Haupt. **[Debugging at Full Speed](http://www.lifl.fr/dyla14/papers/dyla14-3-Debugging_at_Full_Speed.pdf)**. In proceedings of the 8th Workshop on Dynamic Languages and Applications (DYLA), 2014. ([source code](http://lafo.ssw.uni-linz.ac.at/truffle/debugging/dyla14-debugging-artifact-0557a4f756d4.tar.g ...
+<a class="paper" name="seaton2014" href="#seaton2014">#</a> C. Seaton, M. L. Van De Vanter, M. Haupt. **[Debugging at Full Speed](http://www.lifl.fr/dyla14/papers/dyla14-3-Debugging_at_Full_Speed.pdf)**. In proceedings of the 8th Workshop on Dynamic Languages and Applications (DYLA), 2014. ([source code](https://web.archive.org/web/20150117042919/http://lafo.ssw.uni-linz.ac.at/truffle/debugging/d ...
```
I've asked if this can be restored.

arbox replied:
This is still dead.

chrisseaton commented on index.md:
```diff
@@ -257,4 +257,4 @@ In the IPSJ Journal of Programming (PRO), 2005. *In Japanese*.
 ## Standards
 
-<a class="paper" name="iso" href="#iso">#</a> IPA Ruby Standardization WG. **[Programming Languages — Ruby (Draft)](https://www.ipa.go.jp/osc/english/ruby/)**. Final draft of the Ruby ISO standard, 2010.
+<a class="paper" name="iso" href="#iso">#</a> IPA Ruby Standardization WG. **[Programming Languages — Ruby (Draft)](https://web.archive.org/web/20190401051050/https://www.ipa.go.jp/osc/english/ruby/)**. Final draft of the Ruby ISO standard, 2010.
```
This worries me - I'll ask around about where it is now.

chrisseaton replied:
https://twitter.com/ChrisGSeaton/status/1222192770871369728

chrisseaton replied:
It looks like this has been deliberately removed by the authoring body. Since they now only present a paid option, I'm not sure it's right to link to the archived old free version. What do you think?

arbox replied:
As Aaron translated "they don't have it anymore". Difficult to say, but I'd rather delegate the legal questions to Web Archive and change the bibliography link to point there.

If somebody wants some information to disappear the way is to request search engines and archives to delete them. Web Archive deletes data upon request:

```
How can I exclude or remove my site's pages from the Wayback Machine?

You can send an email request for us to review to info@archive.org with the URL (web address) in the text of your message.
```

chrisseaton replied:
Hmmm thanks very much for taking the time to talk me through your reasoning but I'm afraid I don't agree and I'd rather not use archive.org to link to a paper that the original authors have taken down for whatever reason. Could we merge the rest of the PR without that?

arbox replied:
Not a problem :)
Do you want to preserve the bib item without any links? Should I add a link to the official paid version?

chrisseaton replied:
Yes should still link to it - it's not a link of freely available research. We'll probably find more papers in the future that don't have a public version available (we've been lucky so far.) Could link to the paid version, or not link to anything. I think I'd do the former.

chrisseaton replied:
Shall we merge this, then?

chrisseaton commented:
Thanks very much for this.

Can you document the use of `awesome_bot` in the README? I'm not sure how to run it myself. And add it to CI?

arbox committed:
Link to the paid standard version instead of the missing draft

We decided to accept the fact that the free draft/preview version
of the Ruby standard is deleted and should not be taken from
any archived versions. Therefore we'll link to the official final
paid version.

arbox committed:
Document usage of the link linter

We introduced `awesome_bot` and will maintain the correct state
for link sources through regular invocations.

arbox commented:
@chrisseaton ,

I've changed the link to the Ruby ISO Standard and have also described the invocation of the AwesomeBot in the README.
Please look over this PR, I think, it can be merged.

I have not much time so I cannot promise any results in the next days but I'll incorporate `AwesomeBot` into the CI pipeline within a separate PR.

Thank you for maintaining this resource! It's useful :)
